service-course-crawler/src/main/kotlin/de/hpi/cloud/course/crawler/Main.kt

```
@@ -0,0 +1,93 @@
+        .userAgent(USER_AGENT_STRING)
+}
+
+fun main(args: Array<String>) {
+    println("Starting $NAME")
+    withBucket("course") { bucket ->
+        if (args.contains("--clear"))
+            deleteOldData(bucket)
+        else {
+            println("Crawling current semester") // TODO: implement archive crawler
+            println("Using User-Agent=\"$USER_AGENT_STRING\"")
+
+            var count = 0
+            CRAWLERS.asSequence()
+                .flatMap {
+                    println("Starting crawler for $it")
+                    it.listCourses()
+                }
+                .map {
+                    try {
+                        it.query()
+                    } catch (ex: Exception) {
+                        println("error: $ex")
+                        null
+                    }
+                }
+                .filterNotNull()
+                .forEach {
+                    println("Parsed course page with ID ${it.id}")
+                    bucket.upsert(it.toJsonDocument())
+                    bucket.upsert(it.course.toJsonDocument())
+                    bucket.upsert(it.course.courseSeries.toJsonDocument())
+                    count++
+                }
+            println("Upserted $count course page")
+        }
+    }
+    println("Crawler used ${requestCount} server requests")
+}
+
+fun deleteOldData(bucket: Bucket) {
+    println("Entered database cleanup mode\n")
+    bucket.query(ViewQuery.from("course", VIEW_BY_ID))
+        .allRows()
+        .forEach {
+            println("Remove Course ${it.id()}")
+            try {
+                bucket.remove(it.id())
+            } catch (ex: Exception) {
+            }
+        }
+    println()
+    bucket.query(ViewQuery.from("courseSeries", VIEW_BY_ID))
+        .allRows()
+        .forEach {
+            println("Remove CourseSeries ${it.id()}")
+            try {
+                bucket.remove(it.id())
+            } catch (ex: Exception) {
+            }
```

Don't silently ignore exceptions, at least add a comment